indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx

```diff
@@ -0,0 +1,296 @@
+    return dt.toLocaleDateString(undefined, options);
+  }
+
+  const currentSelectionMsg = $T.gettext('You selected {0} session blocks.').format(value.length);
+
+  return (
+    <Form.Group styleName="timetablesessions-field">
+      <strong>{currentSelectionMsg}</strong>
+      {sessionData &&
+        Object.keys(sessionData)?.map((date, index) => (
+          <SessionBlockHeader
+            value={value}
+            index={index}
+            name={name}
+            data={sessionData[date]}
+            label={formatDate(date)}
+            allowFullDays={allowFullDays}
+            display={display}
+            key={`session-block-${date}`}
+            handleValueChange={handleValueChange}
+            onChange={onChange}
+            props={props}
+          />
+        ))}
+    </Form.Group>
+  );
+}
+
+TimetableSessionsComponent.propTypes = {
+  value: PropTypes.arrayOf(PropTypes.number).isRequired,
+  name: PropTypes.string.isRequired,
+  sessionData: PropTypes.objectOf(
+    PropTypes.arrayOf(
+      PropTypes.shape({
+        full_title: PropTypes.string.isRequired,
+        id: PropTypes.number.isRequired,
+        start_dt: PropTypes.string.isRequired,
+      })
+    )
+  ),
+  onChange: PropTypes.func.isRequired,
+  onFocus: PropTypes.func.isRequired,
+  onBlur: PropTypes.func.isRequired,
+  allowFullDays: PropTypes.bool.isRequired,
+  display: PropTypes.oneOf(['expanded', 'collapsed']),
+  maximum: PropTypes.number.isRequired,
+  minimum: PropTypes.number.isRequired,
+};
+
+TimetableSessionsComponent.defaultProps = {
+  sessionData: {},
+};
+
+function SessionBlockHeader({
+  value,
+  data,
+  label,
+  allowFullDays,
+  display,
+  handleValueChange,
+  onChange,
+}) {
+  const childrenIds = data.map(e => e.id);
+  const isChecked = childrenIds.reduce((acum, id) => {
+    return acum && value.includes(id);
+  }, true);
+  const isIndeterminate = childrenIds.reduce((acum, id) => {
+    return acum || value.includes(id);
+  }, false);
+  const [isExpanded, setIsExpanded] = useState(display === 'expanded');
+
+  const isExpandedClass = isExpanded ? '' : 'hidden';
+
+  function handleHeaderClick() {
+    setIsExpanded(!isExpanded);
+  }
+
+  function handleHeaderChange() {
+    const newValue =
+      isChecked || isIndeterminate
+        ? value.filter(element => !childrenIds.includes(element))
+        : [...value, ...childrenIds];
+    onChange(newValue);
+  }
+
+  return (
+    <>
+      <Accordion styleName="equipment-accordion" className="ui-state-default">
+        <Accordion.Title active={isExpanded} index={0} onClick={handleHeaderClick}>
+          <Icon name="dropdown" />
+          <Checkbox
+            checked={isChecked}
+            onChange={handleHeaderChange}
+            indeterminate={isIndeterminate && !isChecked}
+            disabled={!allowFullDays}
+            label={label}
+          />
+        </Accordion.Title>
+      </Accordion>
+      {data
+        .map(item => camelizeKeys(item))
```

Comment:
I'd camelize it in the same place where you receive the response (`useIndicoAxios`)